ql/src/java/org/apache/hadoop/hive/ql/ddl/view/materialized/alter/rebuild/AlterMaterializedViewRebuildAnalyzer.java

```diff
@@ -480,8 +460,6 @@ protected ASTNode fixUpAfterCbo(ASTNode originalAst, ASTNode newAst, CalcitePlan
         fixUpASTAggregateInsertDeleteIncrementalRebuild(fixedAST, getMaterializedViewASTBuilder());
         return fixedAST;
       case JOIN_INSERT_DELETE_REBUILD:
-        fixUpASTJoinInsertDeleteIncrementalRebuild(fixedAST, getMaterializedViewASTBuilder());
-        return fixedAST;
```

Comment:
This will end up throwing the UnsupportedOperationException.  Don't we want to fall back to the full rebuild ?

Reply:
At this point CBO plan is already transformed to AST hence we can not fall back. However `JOIN_INSERT_DELETE_REBUILD` is never set in the code any more.

Reply:
Ok, just for my reference, for the non-groupby queries that this patch is addressing, could you point to where the fall back to the full rebuild (using insert-overwrite) happens ?

Reply:
Removed.